integration_tests/sdk/data_integration_tests/s3_test.py

```diff
@@ -292,6 +295,53 @@ def test_s3_fetch_directory_with_merge(flow_manager, data_integration):
     assert expected_merged_data.equals(dir_contents_merged.get())
 
 
+def test_s3_fetch_directory_with_delete(flow_manager, data_integration):
+    """Create a random directory name and save a table artifact into it. Delete the workflow,
+    and include that object in the deletion. After that check if the directory is empty."""
+    dir_name = generate_object_name()
+
+    # Order hotel_reviews to be listed before customers by ordering the paths alphabetically.
+    hotel_reviews_table_name = generate_table_name()
+
+    # Write a tabular artifact into the directory.
+    hotel_reviews = data_integration.file(
+        "hotel_reviews", artifact_type=ArtifactType.TABLE, format="parquet"
+    )
+    flow = _save_artifact_and_check(
+        flow_manager,
+        data_integration,
+        artifact=hotel_reviews,
+        format="parquet",
+        object_identifier="%s/%s" % (dir_name, hotel_reviews_table_name),
+    )
+
+    # Check that the artifact can be fetched by directory search.
+    dir_contents = data_integration.file(
+        dir_name + "/",
+        artifact_type=ArtifactType.TABLE,
+        format="parquet",
+    )
+    assert dir_contents.type() == ArtifactType.TUPLE
+    assert dir_contents.get()[0].equals(hotel_reviews.get())
+
+    # Delete the workflow and include artifact into deletion.
+    tables = flow.list_saved_objects()
+    flow_manager._client.delete_flow(
+        flow_id=str(flow.id()),
+        saved_objects_to_delete=tables,
+        force=True,
+    )
+    # Check if the directory is an empty directory.
```

Comment:
This is the unexpected behavior right? If behavior is unexpected let's make a task for it and document that task here with `#TODO(ENG-....): `